test/rules/end_doc_comments_with_punctuation.dart

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2020, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N end_doc_comments_with_punctuation`
+
+// BAD
+
+// LINT [+1]
+/// This sentence doesn't have any terminating punctuation at the end
+void a() => null;
+
+// LINT [+2]
+/// This is a long documentation comment composed of multiple sentences.
+/// However, it looks like we forgot to terminate the last one, which is bad
+void b() => null;
+
+// LINT [+4]
+// ignore: slash_for_doc_comments
+/**
+ * We also flag Java-style documentation comments, such as this long one we have
+ * spread across multiple lines here
+ */
+void c() => null;
+
+// GOOD
+
+// OK
+/// This sentence is properly terminated with a period.
+void d() => null;
+
+// OK
+/// This is a very long sentence spread across multiple lines of a documentation
+/// comment, but it still ends with terminating punctuation!
```

Comment:
Can you add an example of how a multi-line doc comment could be ignored with `// ignore:`? (See [this guide](https://dart.dev/guides/language/analysis-options#suppressing-rules-for-a-line-of-code) for explanation)

PS: I don't think it can be. But I'm not certain.

Reply:
Ah, you're absolutely right, it looks like `//ignore: xyz` currently only works for the [current or next line](https://github.com/dart-lang/sdk/blob/master/pkg/analyzer/lib/src/ignore_comments/ignore_info.dart#L77).

If the other issues with this rule are resolvable, I think this could possibly be mitigated by either raising the lint for the entire offending doc comment block rather than just the final character, or by ignoring it for an entire file with `//ignore_for_file`. However, neither of those seem like great options.